Examples/Detectors/ContextualDetector/src/PrealignedDetector.cpp

```diff
@@ -0,0 +1,169 @@
+  opt.add_options()(
+      "prealign-seed",
+      boost::program_options::value<size_t>()->default_value(1324354657),
+      "Seed for the decorator random numbers.")(
+      "prealign-iovs",
+      boost::program_options::value<size_t>()->default_value(25),
+      "Number of a valid IOVs.")(
+      "prealign-sigma-iplane",
+      boost::program_options::value<double>()->default_value(100.),
+      "Sigma of the in-plane misalignment in [um]")(
+      "prealign-sigma-oplane",
+      boost::program_options::value<double>()->default_value(50.),
+      "Sigma of the out-of-plane misalignment in [um]")(
+      "prealign-sigma-irot",
+      boost::program_options::value<double>()->default_value(20.),
+      "Sigma of the in-plane rotation misalignment in [mrad]")(
+      "prealign-sigma-orot",
+      boost::program_options::value<double>()->default_value(0.),
+      "Sigma of the out-of-plane rotation misalignment in [mrad]")(
+      "prealign-loglevel",
+      boost::program_options::value<size_t>()->default_value(3),
+      "Output log level of the alignment decorator.")(
+      "prealign-firstnominal",
+      boost::program_options::value<bool>()->default_value(false),
+      "Keep the first iov batch nominal.");
+}
+
+auto PrealignedDetector::finalize(
+    const boost::program_options::variables_map& vm,
+    std::shared_ptr<const Acts::IMaterialDecorator> mdecorator)
+    -> std::pair<TrackingGeometryPtr, ContextDecorators> {
+  // --------------------------------------------------------------------------------
+  DetectorElement::ContextType nominalContext;
+
+  auto buildLevel = vm["geo-generic-buildlevel"].template as<size_t>();
+  // set geometry building logging level
+  Acts::Logging::Level surfaceLogLevel =
+      Acts::Logging::Level(vm["geo-surface-loglevel"].template as<size_t>());
+  Acts::Logging::Level layerLogLevel =
+      Acts::Logging::Level(vm["geo-layer-loglevel"].template as<size_t>());
+  Acts::Logging::Level volumeLogLevel =
+      Acts::Logging::Level(vm["geo-volume-loglevel"].template as<size_t>());
+
+  bool buildProto =
+      (vm["mat-input-type"].template as<std::string>() == "proto");
+
+  /// return the generic detector - with aligned context decorator
+  TrackingGeometryPtr aTrackingGeometry =
+      ActsExamples::Generic::buildDetector<DetectorElement>(
+          nominalContext, detectorStore, buildLevel, std::move(mdecorator),
+          buildProto, surfaceLogLevel, layerLogLevel, volumeLogLevel);
+
+  Acts::Logging::Level decoratorLogLevel =
+      Acts::Logging::Level(vm["prealign-loglevel"].template as<size_t>());
+
+  // Let's create a reandom number service
+  ActsExamples::RandomNumbers::Config randomNumberConfig;
+  randomNumberConfig.seed = vm["prealign-seed"].template as<size_t>();
+  auto randomNumberSvc =
+      std::make_shared<ActsExamples::RandomNumbers>(randomNumberConfig);
+
+  // Alignment decorator service
+  Decorator::Config pagcsConfig;
+  pagcsConfig.detectorStore = detectorStore;
```

Comment:
Same here. It might not be necessary any more.

Reply:
It is, the detector store needs to be alive.